### ISSUE TITLE
Textbook Images and diagrams not captured

### ISSUE THREAD

**Issue by Plane3** [erikas-taroza/redshelf_downloader#8]
Hi there,
I successfully downloaded a copy of my textbook and as best as I can tell all the formatting seems to be correct, however I noticed that it failed to import the images and graphs when it imported the text and formatting, there are still little boxes where the images should be but no image. Is there a way to fix this? I will attach a copy of the file incase a example would be helpful.
[result.pdf](https://github.com/user-attachments/files/17742609/result.pdf)

 I have very little experience with this sort of thing so it could be a simple lack of understanding on my part.
 
 Thank you for your help, this is a really helpful tool!

**Comment by erikas-taroza** [comment]
If you go to the `pages` folder, do any of the subfolders have images in them?

**Comment by Plane3** [comment]
yes, most of them do appear to have a subfolder that is labeled images in them however when I try to open one of the PNG or JPG files they contain I have not been able to successfully manage it. Every program I try either says that it doesn't support the format or there was a problem opening the file... I will attach a couple of the files for you.
![13](https://github.com/user-attachments/assets/2bcabaa7-0bf0-4e1e-85a7-ef5505843857)
![78](https://github.com/user-attachments/assets/ec9ef05e-81f3-4ada-b0ac-500b7687c5ee)
![79](https://github.com/user-attachments/assets/7335dae4-2f41-42e4-bd3e-83007e813093)
![80](https://github.com/user-attachments/assets/202ae542-aaeb-4d3c-a52c-9773802e3e93)
![F03002](https://github.com/user-attachments/assets/39b7f6fc-5ec6-48d1-8926-0dccddacdd56)
![F03004](https://github.com/user-attachments/assets/601c03e4-2261-43df-9bf2-3c0f3960ce71)
![F03005](https://github.com/user-attachments/assets/115c7d92-8730-47c9-9379-78beea5b8d56)
![F03006](https://github.com/user-attachments/assets/191d6ab3-3cc0-4e43-84c3-31deeaa3388d)
![F03012](https://github.com/user-attachments/assets/e1948d5a-7cca-4d91-b596-1a4f3993154b)
![F03013](https://github.com/user-attachments/assets/e5833524-0af4-4bc5-9b55-557eb928cb2e)
![F03014](https://github.com/user-attachments/assets/b2e0cdfb-f789-45b2-a59f-b610330399e4)
![F03015](https://github.com/user-attachments/assets/21f144e1-df97-4001-acdb-7df4dd5adf0b)

**Comment by erikas-taroza** [comment]
What is the file size of the images?

**Comment by Plane3** [comment]
> What is the file size of the images?

The image files are all exactly 108 KB it was the same for every image folder I looked in

**Comment by erikas-taroza** [comment]
Sounds like something went wrong with the download. Maybe the url is incorrect?

Unfortunately, I cannot test this since I only have 1 book and it works for that book. You can add print statements to see what the url looks like. Also, you could try seeing how the image requests are made by the website and comparing that to what the code is doing.

Make sure you are getting your cookies from the request in the browser with this url format:
`https://platform.virdocs.com/read/document/book_id/page_number`

To see what the url looks like, you can add this code on line 88:

```diff
def download_remote_resources(page: int, base_url: str, urls: list[str]):
    session = create_session()
    path = f"{PAGE_PATH}/{page}"

    for url in urls:
        request_url = url
        # Static is a universal file not dependent on the book.
        if "/static" in url:
            request_url = f"https://platform.virdocs.com{url}"
        else:
            request_url = f"{base_url}{url}"

        response = session.get(request_url, allow_redirects=True, cookies=COOKIES)
        file = Path(f"{path}{url}")
        file.parent.mkdir(parents=True, exist_ok=True)
        file.write_bytes(response.content)
+       print(request_url)


def create_html_file(page: int, raw: str):
```

**Comment by thisismyburner** [comment]
@erikas-taroza, I'm having a similar issue, though images are successfully downloading for me. 
`pages/30/img/chapter02/lar60886_p0201.png`
![Image](https://github.com/user-attachments/assets/07cfd336-68fe-499e-98a4-f802aebe32af)

--

RedShelf vs. result.pdf

![Image](https://github.com/user-attachments/assets/cd457ef4-5412-4efc-903c-dbc5ac1c2e8a)

![Image](https://github.com/user-attachments/assets/5845d007-1b6d-41d4-95f6-22f5c44a2288)

**Comment by thisismyburner** [comment]
Fixed it on my end. Working on a PR!
